Mage/src/main/java/mage/abilities/common/OneOrMoreDealDamageTriggeredAbility.java

```diff
@@ -59,12 +61,12 @@ public boolean checkEventType(GameEvent event, Game game) {
     }
 
     @Override
-    public boolean checkTrigger(GameEvent event, Game game) {
+    public Stream<DamagedPlayerEvent> filterBatchEvent(GameEvent event, Game game) {
         DamagedBatchForOnePlayerEvent dEvent = (DamagedBatchForOnePlayerEvent) event;
         if (onlyCombat && !dEvent.isCombatDamage()) {
```

Comment:
wouldn't it make more sense for these sorts of checks (on info about the batch event rather than the sub events) to be in `checkTrigger` before calling this method for the stream?

Reply:
Yeah I'll do a cleanup pass for the combat damage, and the target (resp. source) when the Batch event is one that has a shared target (resp. source)